src/Microsoft.ApiDesignGuidelines.Analyzers/Core/OperatorOverloadsHaveNamedAlternates.Fixer.cs

```diff
@@ -27,10 +32,126 @@ public sealed override FixAllProvider GetFixAllProvider()
         }
 
         public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
-        {                              
-            // This is to get rid of warning CS1998, please remove when implementing this analyzer
-            await new Task(() => { });
-            throw new NotImplementedException();
+        {
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            var node = root.FindNode(context.Span);
+            var semanticModel = await context.Document.GetSemanticModelAsync(context.CancellationToken).ConfigureAwait(false);
+            var diagnostic = context.Diagnostics.First();
+            switch (diagnostic.Properties[OperatorOverloadsHaveNamedAlternatesAnalyzer.DiagnosticKindText])
+            {
+                case OperatorOverloadsHaveNamedAlternatesAnalyzer.AddAlternateText:
+                    AddFriendlyAlternate(context, root, node, semanticModel, diagnostic);
+                    break;
+                case OperatorOverloadsHaveNamedAlternatesAnalyzer.FixVisibilityText:
+                    await FixFriendlyVisibility(context, node, semanticModel, diagnostic);
+                    break;
+            }
+        }
+
+        private static void AddFriendlyAlternate(CodeFixContext context, SyntaxNode root, SyntaxNode node, SemanticModel semanticModel, Diagnostic diagnostic)
+        {
+            var operatorOverload = (SyntaxNode)node.FirstAncestorOrSelf<TOperatorDeclarationSyntax>() ?? node.FirstAncestorOrSelf<TConversionOperatorSyntax>();
```

Comment:
you can simply call `syntaxGenerator.GetDeclaration(node, DeclarationKind.Operator)` to get the enclosing operator declaration. This will avoid the need for the type parameter. If we can eliminate all type params this way then we don't need the C#\VB specific types at all.